### ISSUE TITLE
`ruff format` or `ruff check` returns "nested alternate groups are not allowed" error

### ISSUE THREAD

**Issue by niccolomineo** [astral-sh/ruff#9381]
I am attempting to exclude this dir from inspection:

```
extend-exclude = [
    "__pycache__",
]
```

but I am getting:

```sh
$ python3 -m ruff format .
ruff failed
  Cause: error parsing glob '/Users/me/project/{{cookiecutter.project_dirname}}/__pycache__': nested alternate groups are not allowed
```

Any ideas?

On a side note, according to [this](https://regex101.com/r/I8WJGr/1) experiment, this syntax should be valid

```
extend-exclude = [
    "[{][{]cookiecutter[.]project_dirname[}][}]/__pycache__",
]
```

but it still gets me the error above.

**Comment by charliermarsh** [comment]
To clarify, the whole config is:

```toml
extend-exclude = [
    "__pycache__",
]
```

Is that right? And you're running from the `/Users/me/project/{{cookiecutter.project_dirname}}` directory?

**Comment by niccolomineo** [comment]
The whole config is:

```
extend-exclude = [
    "__pycache__",
    ".vscode*",
]
```

and I am running from `Users/me/project/`. 

`{{cookiecutter.project_dirname}}` is a subdir in that path.

**Comment by charliermarsh** [comment]
Got it, thanks!

**Comment by niccolomineo** [comment]
> Got it, thanks!

Thank YOU! Are you able to confirm it is a bug and not something I have to tweak on my side?

**Comment by charliermarsh** [comment]
From the description it looks like a bug. What you could do though is omit that directory entirely:

```toml
extend-exclude = [
    "[{][{]cookiecutter.project_dirname[}][}]",
]
```

**Comment by charliermarsh** [comment]
(Taken from: https://github.com/astral-sh/ruff/issues/7959#issuecomment-1764751734)

**Comment by niccolomineo** [comment]
> From the description it looks like a bug. What you could do though is omit that directory entirely:
> 
> 
> 
> ```toml
> 
> extend-exclude = [
> 
>     "[{][{]cookiecutter.project_dirname[}][}]",
> 
> ]
> 
> ```

Unfortunately I need to target specific subdirs of that unusual dir, so that's not viable for me.

**Comment by niccolomineo** [comment]
Hi @charliermarsh, you might be already aware of this, but this is occurring with the base `ruff` command as well, not just with `ruff format`.

**Comment by GitRon** [comment]
We have the same issue in combination with cookiecutter 😕 The curly brackets seem to be a problem.

**Comment by knowfahad** [comment]
Any update on this?

**Comment by lucasfijen** [comment]
Yep I'm running in exactly the same in combination with cookiecutter.

I have a folder structure:

{{ cookiecutter.project_namel }}/
├─ pyproject.toml
├─ tests/
│  ├─ maintest.py
├─ main.py

Where my pyproject.toml contains this setting that is causing the issue, removing this line fixes the error

```pyproject.toml
[tool.ruff.lint.per-file-ignores]
"tests/*" = ["F811"]
```

Then running ruff from the root  using `cookiecutter format .` will cause:
```
ruff failed
  Cause: error parsing glob '/workspaces/reponame/{{cookiecutter.repo_name}}/tests/*': nested alternate groups are not allowed
```

**Comment by ntBre** [comment]
Does anyone have a link to a repo or gist where I could reproduce this? I'm trying to take a look but having some trouble reproducing it locally with this file tree:

```
.
├── ruff.toml
└── {{cookiecutter.project_dirname}}
    ├── __pycache__
    │   └── try.py
    └── try.py
```

We use [globset](https://docs.rs/globset/latest/globset/) for most of our path globs, which has the comment "Nesting {...} is not currently allowed" in the docs, so I think the curly braces are definitely the problem.

For now the best workarounds are still "escaping" the curly braces with square brackets (`[{]`) or backslashes (`\{`, if you're not on Windows), but I agree that it would be nice if we had some heuristics for detecting this automatically.

**Comment by lucasfijen** [comment]
Thank you for taking your time to looking into it! This monday I should have some time to setup a small example repo for you to make it reproducible. 

The issue with this cookiecutter example is that the glob part with the curly brackets is not included in the ignore list field, as that part should be relative paths. However the curly brackets part are included when running ruff from the root (for this example we might need a second pyproject.toml above the ccokiecutter.project_dirname to get it reproduced.

This explanation might not be very clear, I'll work on an example repo for you:)

**Comment by lucasfijen** [comment]
@ntBre as promised I created a repository [here](https://github.com/lucasfijen/example_ruff_glob_bug/settings).

**Comment by ntBre** [comment]
@lucasfijen Thank you so much! I cloned the repo and can now reproduce!
